web/src/ResourceGroupsContext.tsx

```diff
@@ -0,0 +1,69 @@
+import { createContext, PropsWithChildren, useContext } from "react"
+import { AnalyticsAction, AnalyticsType, incr } from "./analytics"
+import { usePersistentState } from "./LocalStorage"
+
+const GROUPS_LOCAL_STORAGE_NAME = "resource-groups"
+const DEFAULT_GROUP_STATE = true
+
+export type GroupExpandedState = { [key: string]: boolean }
+
+type ResourceGroupsContext = {
+  expanded: GroupExpandedState
+  setGroup: (groupLabel: string, page: AnalyticsType) => void
+  getGroup: (groupLabel: string) => boolean
+}
+
+const resourceGroupsContext = createContext<ResourceGroupsContext>({
+  expanded: {},
+  setGroup: () => {
+    // Note: this warning should only show in development
+    console.warn(
+      "Resource group context is not set. Did you forget to provide it?"
+    )
+  },
+  getGroup: () => {
+    // Note: this warning should only show in development
+    console.warn(
+      "Resource group context is not set. Did you forget to provide it?"
+    )
```

Comment:
I don't have strong opinions about keeping this warning message, but I found it useful in my development.